### PR TITLE
update README.md, fix `make test`, add circle.yml

### PR DESCRIPTION
**Pull request by tiancaiamao**
* Remove go.mod so the TiDB and parser repo won't be a circle
* Add circle.yml so this repo will run CI later
* Add README.md about "How to update parser for TiDB"

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: golang:1.11
+    working_directory: /go/src/github.com/pingcap/parser
+    steps:
+      - checkout
+      - run:
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply circle.yml]: That's a good suggestion. 
I'm not familiar with writing scripts however.
Would like to pull a request for it?
@kennytm

[the remark this answers is not in the document]
Comment (kennytm) [reply circle.yml]: @tiancaiamao I create a PR to your branch?

[the remark this answers is not in the document]
Comment (tiancaiamao) [reply circle.yml]: After this PR, maybe.

**Comment by tiancaiamao** [comment]
PTAL @lysu @zimulala

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,11 @@ LINUX:="Linux"
 all: parser.go
 
 test: parser.go
+	mv go.mod1 go.mod
+	mv go.sum1 go.sum
 	GO111MODULE=on go test ./...
+	mv go.mod go.mod1
+	mv go.sum go.sum1
```
Comment (lysu) [inline Makefile]: how about go test failured situation?  it seem re-run go test will meet error.

**Comment by lysu** [comment]
@CaitinChen PTAL

File: Makefile [hunk 3]
```diff
@@ -7,6 +7,9 @@ LINUX:="Linux"
 all: parser.go
 
 test: parser.go
+	mv go.mod1 go.mod
+	mv go.sum1 go.sum
+	trap 'mv go.mod go.mod1;mv go.sum go.sum1' EXIT
```
Comment (lysu) [inline Makefile]: from https://stackoverflow.com/questions/15656492/linux-shell-try-catch-finally it works @tiancaiamao

File: Makefile (hunk 2 above)
Comment (tiancaiamao) [reply Makefile]: done

File: Makefile (hunk 3 above)
Comment (tiancaiamao) [reply Makefile]: I follow the way of the first answer.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,52 @@
...
+
+In your TiDB repository, modify the `go.mod` file, remove `github.com/pingcap/parser` from the `require` instruction, add a new line at the end of file like this:
+
+```
+replace github.com/pingcap/parser => github.com/your-repo/parser v0.0.0-20181102150703-4acd198f5092
+```
+
+This change tells TiDB to use the modified parser from your repository.
+
+You can get correct version information by running this command in your TiDB directory:
```
Comment (CaitinChen) [inline README.md]:
```suggestion
2. You can get correct version information by running this command in your TiDB directory:
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,52 @@
...
+
+```
+GO111MODULE=on go get -u github.com/your-repo/parser@master
+```
+
+If you get some error output, never mind. You can still edit the `go.mod` file manually.
+
+Pull a request to TiDB.
```
Comment (CaitinChen) [inline README.md]:
```suggestion
3. File a PR to TiDB.
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,52 @@
 # parser
+
 TiDB SQL Parser
+
+## How to update parser for TiDB
+
+Suppose you want to pull a request to TiDB, and the PR needs to change the parser, here is how to do it.
```
[the remark this answers is not in the document]
Comment (CaitinChen) [reply README.md]: We usually say "file a PR".

File: README.md [hunk 7]
```diff
@@ -1,2 +1,52 @@
...
+
+```
+GO111MODULE=on go get -u github.com/your-repo/parser@master
+```
+
+If you get some error output, never mind. You can still edit the `go.mod` file manually.
+
+Pull a request to TiDB.
+
+### Step 3, merge the parser to this repository
```
Comment (CaitinChen) [inline README.md]:
```suggestion
### Step 3: Merge the PR about the parser to this repository
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,52 @@
...
+
+Pull a request to this repository. **Link the related PR in TiDB in your PR description or comment**.
```
Comment (CaitinChen) [inline README.md]:
```suggestion
File a PR to this repository. **Link the related PR in TiDB in your PR description or comment.**
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,52 @@
...
+
+The PR will be reviewed, if everything goes well, it will be merged.
```
Comment (CaitinChen) [inline README.md]:
```suggestion
This PR will be reviewed, and if everything goes well, it will be merged.
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,52 @@
...
+
+### Step 4, update TiDB to use the latest parser
```
Comment (CaitinChen) [inline README.md]:
```suggestion
### Step 4: Update TiDB to use the latest parser
```

File: README.md [hunk 11]
```diff
@@ -1,2 +1,52 @@
...
+
+In your TiDB pull request, modify the `go.mod` file manually or use this command:
+
+```
+GO111MODULE=on go get -u github.com/pingcap/parser@master
+```
+
+Make sure the `replace` instruction is changed back to the `require` instruction and the version is latest.
```
Comment (CaitinChen) [inline README.md]:
```suggestion
Make sure the `replace` instruction is changed back to the `require` instruction and the version is the latest.
```

File: README.md [hunk 12]
```diff
@@ -1,2 +1,55 @@
...
+
+### Step 1: Make changes in your parser repository
+
+Fork this repository to your own account and commit the changes to your repository.
+> **Note:**
+>
+> - Don't forget to run `make test` before you commit!
+> - Make sure `parser.go` is updated.
+* Don't forget to run `make test` before you commit!
+* Make sure the `parser.go` is updated.
```
Comment (kennytm) [inline README.md]: these are repeated now

Comment (tiancaiamao) [reply README.md]: Done

**Pull request merged**